drivers/sensor/amg88xx/amg88xx.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2017 Phytec Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <device.h>
+#include <i2c.h>
+#include <gpio.h>
+#include <misc/byteorder.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <sensor.h>
+#include <misc/__assert.h>
+
+#include "amg88xx.h"
+
+int amg88xx_reg_read(struct amg88xx_data *drv_data, u8_t reg, u8_t *val)
```

Comment:
I think these wrapper functions are overkill. We're also beginning to see stack overflows due to really deep call stacks.